### PR TITLE
Don't ignore empty MO translation files

### PR DESCRIPTION
**Pull request by berarma**
Issue #10964.

### CODE REVIEW

File: src/I18n/MessagesFileLoader.php [hunk 1]
```diff
@@ -126,7 +125,7 @@ public function __invoke()
         }
 
         if (!$file) {
-            return $package;
+            return false;
```
Comment (markstory) [inline src/I18n/MessagesFileLoader.php]: Will consumers of this class be ok with getting `false`? I think that's why the old code returned an empty `Package` instance.

File: tests/TestCase/I18n/MessagesFileLoaderTest.php [hunk 2]
```diff
@@ -64,4 +64,15 @@ public function testCustomLocalePath()
...
+
+    /**
+     * Test reading MO files
+     * @return void
+     */
+    public function testMoFiles()
+    {
+        $loader = new MessagesFileLoader('empty', 'es', 'mo');
+        $package = $loader();
+        $this->assertNotNull($package);
```
Comment (markstory) [inline tests/TestCase/I18n/MessagesFileLoaderTest.php]: `false` is not null too. Are there any other assertions we can make about this package object?

File: src/I18n/MessagesFileLoader.php (hunk 1 above)
Comment (berarma) [inline src/I18n/MessagesFileLoader.php]: If that's true then I don't know how to deal with non-existing files. It's perfectly fine for a package to be empty.

File: tests/TestCase/I18n/MessagesFileLoaderTest.php (hunk 2 above)
Comment (berarma) [inline tests/TestCase/I18n/MessagesFileLoaderTest.php]: My bad. I should check for false.

File: src/I18n/MessagesFileLoader.php (hunk 1 above)
Comment (markstory) [inline src/I18n/MessagesFileLoader.php]: Is the empty package causing the cascade to the PO file?

Comment (berarma) [inline src/I18n/MessagesFileLoader.php]: Yes. I need some way to differentiate when the file doesn't exists or doesn't have any translations.

**Comment by berarma** [comment]
I've changed the assertions in the tests and updated the docblock for the modified method.

It doesn't seem the ```MessagesFileLoader``` class is used anywhere else than the ```TranslationRegistry``` class.

The ```ChainMessagesLoader``` class that invokes the class already does a fuzzy check for a false return value, thus I thought it would OK to use false as return value when the file doesn't exist. If a Package should always be expected then that check would be useless.

I could change the return value to null if that makes more sense.

**Comment by markstory** [comment]
I took a look and you're right about the `!$package` check. Given that exists, returning false is reasonable.

**Pull request merged**